CMakeLists.txt

```diff
@@ -28,6 +28,25 @@ if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
         message(FATAL_ERROR "g++ version must be at least 5.1!")
     endif()
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+    include_directories(BEFORE "/opt/intel/include")
+    add_definitions(-D__PURE_INTEL_C99_HEADERS__)
+    link_directories(BEFORE "/opt/intel/lib")
+    if((CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 17) AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19))
+        # locate best gcc version to pass to icc-18 (gcc-8 doesn't work, for example)
+        # Use -DICC_GCCVER=6 for example to force 'gcc-6' and 'g++-6'
+        # Otherwise, the newest available and compatible pair will be selected
+        find_program(ICC_GXX NAMES "g++-${ICC_GCCVER}" "g++-7" "g++-6" "g++")
+        get_filename_component(ICC_GXX ${ICC_GXX} NAME)
+        find_program(ICC_GCC NAMES "gcc-${ICC_GCCVER}" "gcc-7" "gcc-6" "gcc")
+        get_filename_component(ICC_GCC ${ICC_GCC} NAME)
+        set(ICC_COMPILERS "-gxx-name=${ICC_GXX} -gcc-name=${ICC_GCC}")
```

Comment:
Not sure if this is the correct way, because if I install icc I select already an gcc version. This looks like a workaround.

Reply:
What, where?  I run the installer, remove everything but the C++, it never asks it only seeks whatever `/usr/bin/gcc` points to... system default

I also use this to generate test binaries made with every permutation of ICC+GCC regardless what was configured or what the system defaults are.  So yes, it is a workaround, for having to use the ICC environment at all (I don't use their shell scripts or whatever).  I also don't have ICC in the system path.